File: 2441-largest-positive-integer-that-exists-with-its-negative/2441-largest-positive-integer-that-exists-with-its-negative.java
import java.util.*;

class Solution {
    public int findMaxK(int[] nums) {
        boolean[] exist = new boolean[2001];
        int ans = -1;
        for (int num : nums) {
            if (exist[-num + 1000]) {
                ans = Math.max(ans, Math.abs(num));
            }
            exist[num + 1000] = true;
        }
        return ans;
    }
}

File: 2441-largest-positive-integer-that-exists-with-its-negative/2441-largest-positive-integer-that-exists-with-its-negative.cpp
class Solution {
public:
    int findMaxK(vector<int>& nums) {
        vector<bool>exist(2001,false);
        int ans = -1;
        for(int it:nums){
            if(exist[-it+1000])ans=max(ans,abs(it));
            exist[it+1000]=true;
        }
        return ans;
    }
};